dds/DCPS/transport/shmem/ShmemDataLink.cpp

```diff
@@ -24,27 +27,34 @@ OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
 namespace OpenDDS {
 namespace DCPS {
 
+namespace {
+  const Encoding encoding_unaligned_native(Encoding::KIND_UNALIGNED_CDR);
+}
+
 ShmemDataLink::ShmemDataLink(ShmemTransport& transport)
   : DataLink(transport,
              0,     // priority
              false, // is_loopback,
              false) // is_active
-  , config_(0)
   , send_strategy_(make_rch<ShmemSendStrategy>(this))
   , recv_strategy_(make_rch<ShmemReceiveStrategy>(this))
   , peer_alloc_(0)
+  , reactor_task_(transport.reactor_task())
 {
 }
 
 bool
 ShmemDataLink::open(const std::string& peer_address)
 {
+  ACE_GUARD_RETURN(ACE_Recursive_Thread_Mutex, g, mutex_, false);
+
   peer_address_ = peer_address;
   const ACE_TString name = ACE_TEXT_CHAR_TO_TCHAR(peer_address.c_str());
-  ShmemAllocator::MEMORY_POOL_OPTIONS alloc_opts;
+  ShmemAllocator::MEMORY_POOL_OPTIONS* alloc_opts_ptr = 0;
 
 #ifdef OPENDDS_SHMEM_WINDOWS
-  const bool use_opts = true;
+  ShmemAllocator::MEMORY_POOL_OPTIONS alloc_opts;
+  alloc_opts_ptr = &alloc_opts;
```

Comment:
I had a comment about this somewhere else (maybe in this same PR), but I think this is more complex than necessary.  We know at compile time if the `opts` will be used or not.